pystac_client/item_search.py

```diff
@@ -590,71 +590,24 @@ def matched(self) -> Optional[int]:
             warnings.warn("numberMatched or context.matched not in response")
         return found
 
-    def get_item_collections(self) -> Iterator[ItemCollection]:
-        """DEPRECATED. Use :meth:`ItemSearch.item_collections` instead.
-
-        Yields:
-            ItemCollection : a group of Items matching the search criteria within an
-            ItemCollection
-        """
-        warnings.warn(
-            "get_item_collections() is deprecated, use item_collections() instead",
-            DeprecationWarning,
-        )
-        return self.item_collections()
-
-    def item_collections(self) -> Iterator[ItemCollection]:
-        """Iterator that yields ItemCollection objects.  Each ItemCollection is
-        a page of results from the search.
-
-        Yields:
-            ItemCollection : a group of Items matching the search criteria within an
-            ItemCollection
-        """
-        if isinstance(self._stac_io, StacApiIO):
-            for page in self._stac_io.get_pages(
-                self.url, self.method, self.get_parameters()
-            ):
-                yield ItemCollection.from_dict(
-                    page, preserve_dict=False, root=self.client
-                )
-
-    def get_items(self) -> Iterator[Item]:
-        """DEPRECATED. Use :meth:`ItemSearch.items` instead.
-
-        Yields:
-            Item : each Item matching the search criteria
-        """
-        warnings.warn(
-            "get_items() is deprecated, use items() instead",
-            DeprecationWarning,
-        )
-        return self.items()
+    # ------------------------------------------------------------------------
+    # Result sets
+    # ------------------------------------------------------------------------
+    # By item
 
     def items(self) -> Iterator[Item]:
         """Iterator that yields :class:`pystac.Item` instances for each item matching
-        the given search parameters. Calls
-        :meth:`ItemSearch.item_collections` internally and yields from
-        :attr:`ItemCollection.features <pystac_client.ItemCollection.features>` for
-        each page of results.
+        the given search parameters.
 
         Yields:
             Item : each Item matching the search criteria
         """
-        nitems = 0
-        for item_collection in self.item_collections():
-            for item in item_collection:
-                yield item
-                nitems += 1
-                if self._max_items and nitems >= self._max_items:
-                    return
+        for item in self.items_as_dicts():
+            yield Item.from_dict(item, root=self.client, preserve_dict=False)
```

Comment:
This is a real change, which I can revert if desired.

Previously, `.items()` called `.item_collections()` (now called `.pages())` which would convert an entire page of result in a `pystac.ItemCollection` and then yield results from that IC.

Now we just get an item at a time (from `.items_as_dicts())` and convert each to a `pystac.Item` individual. It seemed a bit more logical, and probably more efficient (unless `ItemCollection` does something fancy I don't know about).

I can split this into it's own PR if you'd like.

Reply:
I like it, but does this change remove the `max_items` upper bound?

Reply:
I'll double check that we have a test for this, but `max_items` should still work since we're going through `self.items_as_dicts` which returns early once we hit max_items.

Reply:
It's not dedicated to max_items, but this is tested in `TestItemSearchQuery.test_query_shortcut_syntax` which calls `list(search.items())` and verifies that the length is equal to max_items.